specification/2.0/README.md

```diff
@@ -3903,67 +3891,230 @@ Application-specific data.
 
-As previously defined
+with the Trowbridge-Reitz/GGX microfacet distribution
 
-`const dielectricSpecular = rgb(0.04, 0.04, 0.04)`
-<br>
-`const black = rgb(0, 0, 0)`
+<img src="https://render.githubusercontent.com/render/math?math=\displaystyle D = \frac{\alpha^2 \, \chi^%2B(N \cdot H)}{\pi ((N \cdot H)^2 (\alpha^2 - 1) %2B 1)^2}">
 
-*c<sub>diff</sub>* = `lerp(baseColor.rgb * (1 - dielectricSpecular.r), black, metallic)`
-<br>
-*F<sub>0</sub>* = `lerp(dieletricSpecular, baseColor.rgb, metallic)`
-<br>
-*&alpha;* = `roughness ^ 2`
+and the separable form of the Smith joint masking-shadowing function
 
-Additionally,  
-*V* is the normalized vector from the shading location to the eye  
-*L* is the normalized vector from the shading location to the light  
-*N* is the surface normal in the same space as the above values  
-*H* is the half vector, where *H* = normalize(*L*+*V*)  
+<img src="https://render.githubusercontent.com/render/math?math=\displaystyle G = \frac{2 \, \left| N \cdot L \right| \, \chi^%2B(H \cdot L)}{\left| N \cdot L \right| %2B \sqrt{\alpha^2 %2B (1 - \alpha^2) (N \cdot L)^2}} \frac{2 \, \left| N \cdot V \right| \, \chi^%2B(H \cdot V)}{\left| N \cdot V \right| %2B \sqrt{\alpha^2 %2B (1 - \alpha^2) (N \cdot V)^2}}">,
 
-The core lighting equation the sample uses is the Schlick BRDF model from [An Inexpensive BRDF Model for Physically-based Rendering](https://www.cs.virginia.edu/~jdl/bib/appearance/analytic%20models/schlick94b.pdf)
+where χ<sup>+</sup>(*x*) denotes the Heaviside function: 1 if *x* > 0 and 0 if *x* ≤ 0. See [Heitz (2014)](#Heitz2014) for a derivation of the formulas.
 
-![](figures/lightingSum.PNG)
+Introducing the visibility function
 
-Below are common implementations for the various terms found in the lighting equation.
+<img src="https://render.githubusercontent.com/render/math?math=\displaystyle V = \frac{G}{4 \, \left| N \cdot L \right| \, \left| N \cdot V \right|}">
 
-### Surface Reflection Ratio (F)
+simplifies the original microfacet BRDF to
 
-**Fresnel Schlick**
+<img src="https://render.githubusercontent.com/render/math?math=\displaystyle \text{MicrofacetBRDF} = V D">
+
+with
+
+<img src="https://render.githubusercontent.com/render/math?math=\displaystyle V = \frac{\, \chi^%2B(H \cdot L)}{\left| N \cdot L\right| %2B \sqrt{\alpha^2 %2B (1 - \alpha^2) (N \cdot L)^2}} \frac{\, \chi^%2B(H \cdot V)}{\left| N \cdot V \right| %2B \sqrt{\alpha^2 %2B (1 - \alpha^2) (N \cdot V)^2}}">.
+
+Thus we have the function
+
+```
+function specular_brdf(α) {
+  return V * D
+}
+```
+
+### Diffuse BRDF
+
+The diffuse reflection `diffuse_brdf(color)` is a Lambertian BRDF
+
+<img src="https://render.githubusercontent.com/render/math?math=\displaystyle \text{LambertianBRDF} = \frac{1}{\pi}">
+
+multiplied with the `color`.
+
+```
+function diffuse_brdf(color) {
+  return (1/pi) * color
```

Comment:
This looks like only the D term; I believe you still need the G term multiplied here. Otherwise your diffuse is independent of light direction, and I don't think you're getting any diffuse reflection from light behind the tangent plane.

Reply:
diffuse BRDF is `albedo/PI` the NdotL factor is to account for projected solid angle and is not part of the BRDF itself.

If however you are referencing with G to the microfacet masking function (i.e. Smith) then that would also be incorrect because G is only defined for a specific microfacet with a specific normal, which is only the case for consideration in specular BRDFs.

The only thing that makes sense here is a Heaviside function of `NdotL` but thats usually omitted because when you render in realtime by evaluating the BRDF at specific points, you take care not to evaluate for points outside the hemisphere (behind the tangent plane), and when you render with Monte Carlo or sampling methods, your sample generator should never produce such samples anyway.

Reply:
Hmm, not sure I agree. I think the microfacet and shadowing model applies just as much to the diffuse as the specular, since shadowing stops grazing light from entering the surface at the place you expect. If you use a Heaviside (step) function instead of a smooth G, all that means is that a point light will produce a hard rather than soft shadow line for the colored reflection, which I don't think is a more accurate render. In any case, you said yourself that at least a Heaviside function is required here (which I would call an approximation of G), so either way I think it should be mentioned.

Reply:
`NdotL` is not part of the BRDF. The BRDF is defined as reflected differential radiance over differential incident irradiance. 

`NdotL` is part of the integral over all incoming light directions (upper hemisphere around the surface normal) which a renderer needs to solve to compute the outgoing radiance. I don't think it should be described here.

The Lambertian BRDF used here is the most simple BRDF possible, it's just a constant. A more realistic diffuse-like BRDF that fits into the microfacet framework is the Oren-Nayar BRDF. In this model, the facets are assumed to be Lambertian (as opposed to perfectly specular in the "typical" glossy microfacet models).

Reply:
Ah, okay, I see my mistake now. So `material = ` means the material's BRDF. I think because I saw some `LdotH` terms in there that we were closer to light output. I think it might be nice to show the BRDF integral somewhere just to make it clear how to get all the way from input light to viewer pixel color. I looked up at the material section that references this appendix, and it doesn't define BRDF at all either.

Reply:
> NdotL is not part of the BRDF. The BRDF is defined as reflected differential radiance over differential incident irradiance.

What I've said but more formally.

> The Lambertian BRDF used here is the most simple BRDF possible, it's just a constant. A more realistic diffuse-like BRDF that fits into the microfacet framework is the Oren-Nayar BRDF. In this model, the facets are assumed to be Lambertian (as opposed to perfectly specular in the "typical" glossy microfacet models).

Also Oren-Nayar assumes the V-cavities model, not GGX's NDF.

AFAIK, Oren-Nayar is not a closed form solution but rather a numerical fit of a simulation of such a microfacet model, correct me if I'm wrong

> Hmm, not sure I agree. I think the microfacet and shadowing model applies just as much to the diffuse as the specular, since shadowing stops grazing light from entering the surface at the place you expect. If you use a Heaviside (step) function instead of a smooth G, all that means is that a point light will produce a hard rather than soft shadow line for the colored reflection, which I don't think is a more accurate render. In any case, you said yourself that at least a Heaviside function is required here (which I would call an approximation of G), so either way I think it should be mentioned.

Like I said, if you assume microfacets to be anything other than perfectly smooth, your average microfacet reflection is not 1/PI anymore...

A perfectly smooth surface has no geometric shadowing, except by the tangent plane.

We can mention a heaviside function, but its not required in the reference implementation because of practicality:
1) No one will call the BRDF evaluation function when the observer is below the tangent plane (you'd be seeing the back face of a triangle, VdotN is negative) * actually for smooth shaded triangles, a `if (NdotV>0.0)` check would make sense.
2) No one will call the BRDF evaluation function when the surface is backfacing towards the light, people usually use a `max(NdotL,0.0)` outside the BRDF or an `if (NdotL>0.0)` before evaluating lighting

Changing the microfacet distribution to V-cavities necessitates you use Oren-Nayar, which already has the geometric shadowing baked into the fit AFAIK.

Using the same GGX distribution as specular introduces its own problems, mainly that such a diffuse BRDF has no closed solution and needs to be stochastically simulated with inifnite random walks like in Heitz's papers. and it seems there is no standardised numerical fit.

Also you need to decide whether the specular and diffuse distributions are in a coating or a fuzed substrate, basically:
1) is there a layer of coating (varnish) that has different microsurface on top and on bottom (uncorrelated microfacets from two layers)
2) are the individual microfacets all the same and exhibit specular and diffuse behaviour (correlated microfacet from pretty much a single layer)

For practicality, all the models I've seen even in offline rendering use option (1) where the two BRDFs are treated as separate physical layers.